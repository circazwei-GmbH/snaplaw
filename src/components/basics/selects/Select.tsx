import React from "react";
import { Picker, PickerItem } from "react-native-woodpicker";
import { StyleSheet, View, ViewStyle } from "react-native";
import { SimpleLineIcons } from "@expo/vector-icons";

type SelectProps = {
  items: Array<{ label: string; value: string }>;
  selectedValue: PickerItem | undefined;
  onValueChange: (itemValue: PickerItem, itemIndex: number) => void;
  style?: ViewStyle | undefined;
};

export default function Select({
  onValueChange,
  selectedValue,
  items,
  style,
}: SelectProps) {
  return (
    <View style={{ ...style, ...styles.select }}>
      <Picker onItemChange={onValueChange} item={selectedValue} items={items} />
      <SimpleLineIcons
        style={styles.icon}
        name="arrow-down"
        size={14}
        color="#668395"
      />
    </View>
  );
}

const styles = StyleSheet.create({
  select: {
    backgroundColor: "#EFF7FD",
    height: 44,
    borderRadius: 10,
    paddingHorizontal: 16,
    color: "#202020",
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
  },
  icon: {
    marginLeft: 14,
  },
});
