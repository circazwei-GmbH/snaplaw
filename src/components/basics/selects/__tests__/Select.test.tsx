import React from "react";
import { render } from "@testing-library/react-native";
import Select from "../Select";

const ITEMS = [
  {
    value: "Test",
    label: "Test-label",
  },
];

describe("Select", () => {
  it("Select should display", () => {
    const handler = jest.fn();
    const { toJSON } = render(
      <Select items={ITEMS} selectedValue={ITEMS[0]} onValueChange={handler} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
