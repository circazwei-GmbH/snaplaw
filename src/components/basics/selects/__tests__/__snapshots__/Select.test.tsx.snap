// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Select should display 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#EFF7FD",
      "borderRadius": 10,
      "color": "#202020",
      "flexDirection": "row",
      "height": 44,
      "justifyContent": "center",
      "paddingHorizontal": 16,
    }
  }
>
  <TextInput
    item={
      Object {
        "label": "Test-label",
        "value": "Test",
      }
    }
    items={
      Array [
        Object {
          "label": "Test-label",
          "value": "Test",
        },
      ]
    }
    onItemChange={[MockFunction]}
    placeholder="Test-label"
    testID="Picker"
    value="Test-label"
  />
  <Text />
</View>
`;
